File: JAVA/S1_BASIC/L5_BASICRECURSION/fibonacci.java
class solution{
    int fib(int n){
        if(n==0||n==1)return n;
        return fib(n-1)+fib(n-2);
    }
    int fib2(int num){
        int fibn=0;
        int fib0=0;
        int fib1=1;
        for(int i=2;i<=num;i++){
            fibn=fib0+fib1;
            fib0=fib1;
            fib1=fibn;
            

        }
        return fibn;
    }
}
public class fibonacci{
    public static void main(String[] args){
        solution s=new solution();
        System.out.print(s.fib2(7));
    }
}

File: python/S1_BASIC/L5_BASICRECURSION/fibonacci.py
class Fibonacci:
    def __init__(self,n):
        self.n=n
    def fib(self,n):
        if n<0:
            return -1
        if n==0 or n==1:
            return n
        return self.fib(n-1)+self.fib(n-2)
if __name__=="__main__":
    n=int(input())
    f=Fibonacci(n)
    res=f.fib(f.n)
    print(res)

